openpype/hosts/maya/plugins/publish/extract_fbx_animation.py

```diff
@@ -0,0 +1,68 @@
+# -*- coding: utf-8 -*-
+import os
+
+from maya import cmds  # noqa
+import pyblish.api
+
+from openpype.pipeline import publish
+from openpype.hosts.maya.api import fbx
+
+
+class ExtractFBXAnimation(publish.Extractor):
+    """Extract Rig in FBX format from Maya.
+
+    This extracts the rig in fbx with the constraints
+    and referenced asset content included.
+    This also optionally extract animated rig in fbx with
+    geometries included.
+
+    """
+    order = pyblish.api.ExtractorOrder
+    label = "Extract Animation (FBX)"
+    hosts = ["maya"]
+    families = ["animation.fbx"]
+
+    def process(self, instance):
+        # Define output path
+        staging_dir = self.staging_dir(instance)
+        filename = "{0}.fbx".format(instance.name)
+        path = os.path.join(staging_dir, filename)
+
+        # The export requires forward slashes because we need
+        # to format it into a string in a mel expression
+        fbx_exporter = fbx.FBXExtractor(log=self.log)
+        out_set = instance.data.get("animated_skeleton", [])
+        # Export
+        instance.data["constraints"] = True
+        instance.data["skeletonDefinitions"] = True
+        instance.data["referencedAssetsContent"] = True
+
+        fbx_exporter.set_options_from_instance(instance)
+
+        out_set_name = next(out for out in out_set)
+        # temporarily disable namespace
+        namespace = out_set_name.split(":")[0]
+        new_out_set = out_set_name.replace(
+            f"{namespace}:", "")
+        cmds.namespace(set=':')
+        cmds.namespace(set=namespace)
+        cmds.namespace(rel=True)
+
+        fbx_exporter.export(
+            new_out_set, path.replace("\\", "/"))
+        # restore namespace after export
+        cmds.namespace(set=':')
+        cmds.namespace(rel=False)
+
+        if "representations" not in instance.data:
```

Comment:
This could be:
```py
representations = instance.data.setdefault("representations", [])
representations.append({
            'name': 'fbx',
            'ext': 'fbx',
            'files': filename,
            "stagingDir": staging_dir
})
```

Reply:
i didn't change this t as we need to extract animation in abc  before extract fbx animation.

Reply:
The recommended change by @Minkiu would behave exactly the same as your current logic.

`dict.setdefault` set it to the value only if it does not exist, else it returns the value - and thus you'd actually be appending to the `instance.data["representations"]` that already existed. So the example code provided is still valid even for your use case.